src/test/java/wolox/training/controllersTest/BookControllerTest.java

```diff
@@ -0,0 +1,168 @@
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import org.junit.jupiter.api.Test;
+import java.util.ArrayList;
+
+@RunWith(SpringRunner.class)
+@WebMvcTest(controllers = BookController.class)
+public class BookControllerTest {
+
+    @Autowired
+    private MockMvc mvc;
+
+    @MockBean
+    private BookRepository bookRepository;
+
+    private static Book bookTest;
+    private static List<Book> bookTests;
+    private static final String PATH = "/api/books";
+
+    @BeforeAll
+    static void setUp() {
+        bookTests = EntitiesTest.mockManyBooks();
+        bookTest = EntitiesTest.mockBook();
+
+    }
+
+    @WithMockUser(value = "pablo")
+    @Test
+    @DisplayName("Test find all book ,return status OK")
+    void whenFindBookByIdThenReturnStatusOK() throws Exception {
+        given(bookRepository.findById(1L)).withFailMessage(ErrorConstants.SUCCESS_CORRECT);
+        String url = (PATH + "/1");
+        mvc.perform(get(url)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk());
+    }
+
+    @WithMockUser(value = "pablo")
+    @Test
+    @DisplayName("Test, book is searched for its id,it return status not found")
+    void whenBookThatNotExistsThenReturnNotFound() throws Exception {
+        given(bookRepository.findById(1L)).withFailMessage(ErrorConstants.NOT_EXIST_ID);
+        String url = (PATH + "/1");
+        mvc.perform(get(url)
+                .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isNotFound());
+    }
+
+    @Test
+    @DisplayName("Test , book is created , it return status Created")
+    void whenCreateBookThenReturnStatusCreated() throws Exception {
+        String json = new ObjectMapper().writeValueAsString(bookTest);
+        String url = PATH;
+        mvc.perform(post(url)
+                .contentType(MediaType.APPLICATION_JSON)
+                .characterEncoding("utf-8")
+                .content(json))
+                .andDo(print())
+                .andExpect(status().isCreated());
+    }
+
+    @WithMockUser(value = "pablo")
+    @Test
+    @DisplayName("Test, book is updated , it return status OK")
+    void whenUpdateBookThenReturnStatusCreated() throws Exception {
+        given(bookRepository.findById(1L)).withFailMessage(ErrorConstants.SUCCESS_CORRECT);
+        String json = new ObjectMapper().writeValueAsString(bookTest);
+        String url = (PATH + "/1");
+        mvc.perform(put(url)
+                .contentType(MediaType.APPLICATION_JSON)
+                .characterEncoding("utf-8")
+                .content(json))
+                .andDo(print())
+                .andExpect(status().isOk());
```

Comment:
When we test for an ok status, it would be good to add some validation related to the content of the response

Reply:
I added a new sentence please review and let me know if it is correct.